src/NuGet.Core/NuGet.Protocol/Utility/GetDownloadResultUtility.cs

```diff
@@ -93,6 +93,7 @@ public static class GetDownloadResultUtility
                                     packageStream,
                                     globalPackagesFolder,
                                     downloadContext.ParentId,
+                                    downloadContext.Settings,
```

Comment:
And this is the only other palce which is using `downloadContext.Settings` and here also it doesn't make sense to be there. Instead I'd argue to even have a separate parameter. If we can't have `packageExtractionContext` flowing all the way from top to down like any other context `downloadContext` `nuGetProjectContext`, etc...